src/Processors/Transforms/AggregatingPartialResultTransform.cpp

```diff
@@ -0,0 +1,42 @@
+#include <Processors/Transforms/AggregatingPartialResultTransform.h>
+
+namespace DB
+{
+
+AggregatingPartialResultTransform::AggregatingPartialResultTransform(
+    const Block & input_header, const Block & output_header, AggregatingTransformPtr aggregating_transform_,
+    UInt64 partial_result_limit_, UInt64 partial_result_duration_ms_)
+    : PartialResultTransform(input_header, output_header, partial_result_limit_, partial_result_duration_ms_)
+    , aggregating_transform(std::move(aggregating_transform_))
+    {}
+
+PartialResultTransform::ShaphotResult AggregatingPartialResultTransform::getRealProcessorSnapshot()
+{
+    std::lock_guard lock(aggregating_transform->snapshot_mutex);
+
+    auto & params = aggregating_transform->params;
+    /// Currently not supported cases
+    /// TODO: check that insert results from prepareBlockAndFillWithoutKey return values without changing of the aggregator state
+    if (params->params.keys_size != 0 /// has at least one key for aggregation
```

Comment:
Can we check this condition in `AggregatingTransform::supportPartialResultProcessor` and do not insert `AggregatingPartialResultTransform` at all for this cases?